nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/AnalyzeWorker.cs

```diff
@@ -353,14 +353,15 @@ internal static async Task<bool> AreAllPackagesCompatibleAsync(
         ILogger logger,
         CancellationToken cancellationToken)
     {
-        foreach (var packageId in packageIds)
+        foreach (var packageId in packageIds.SelectMany(id => id.Split(';').Select(part => part.Trim())))
```

Comment:
I'd back off one level and make sure that `packageIds` coming into this function has already been split.